distributed/core.py

```diff
@@ -278,24 +273,19 @@ async def finished(self):
 
     def __await__(self):
         async def _():
-            timeout = getattr(self, "death_timeout", 0)
-            async with self._startup_lock:
-                if self.status in Status.ANY_RUNNING:
-                    return self
-                if timeout:
-                    try:
-                        await asyncio.wait_for(self.start(), timeout=timeout)
-                        self.status = Status.running
-                    except Exception:
-                        await self.close(timeout=1)
-                        raise TimeoutError(
-                            "{} failed to start in {} seconds".format(
-                                type(self).__name__, timeout
-                            )
-                        )
-                else:
-                    await self.start()
-                    self.status = Status.running
+            if self.__status != Status.init:
+                await self._started.wait()
```

Comment:
The worker could have previously failed to start and so this event will never be set

Reply:
I think the `_startup_lock` works better to make concurrent starts wait for the first start